src/site/markdown/examples/annotation-processor.md

```diff
@@ -23,35 +23,38 @@ under the License.
 For example, the [Hibernate Processor](https://hibernate.org/orm/processor/) provides an annotation processor to generate the JPA metamodel.
 
 ## Recommended way to activate annotation processing
-Up to JDK 23 the compiler automatically scanned the classpath for annotation processors and executed all found by default. 
-For security reasons this got disabled by default and annotation processing needs to be activated explicitly.
-The recommended way for this is to list all desired processors using either the `<annotationProcessors>` or the `<annotationProcessorPaths>` configuration.
+Up to JDK 23, the compiler automatically scanned the classpath for annotation processors and executed all found by default.
+For security reasons, this got disabled by default since JDK 23 and annotation processing needs to be activated explicitly.
+The recommended way for this is to list all desired processors using either the `<annotationProcessors>` plugin configuration
+or by declaring the processors as dependencies of type `processor`. `classpath-processor` or `modular-processor`.
 Only those processors will get executed by the compiler.
 
-The following example shows, how to activate the Hibernate Processor.
+The following example shows how to activate the Hibernate Processor.
+Note: this configuration requires Maven 4 with Maven Compiler Plugin 4.
```

Comment:
As not all users will instantly change to Maven 4, should be show the Maven 3 and Maven 4 way in this article?